src/tools/run-make-support/src/drop_bomb/mod.rs

```diff
@@ -0,0 +1,37 @@
+//! This module implements "drop bombs" intended for use by command wrappers to ensure that the
+//! constructed commands are *eventually* executed. This is exactly like `rustc_errors::Diag`
+//! where we force every `Diag` to be consumed or we emit a bug, but we panic instead.
+//!
+//! This is inspired by <https://docs.rs/drop_bomb/latest/drop_bomb/>.
+
+use std::borrow::Cow;
+
+#[cfg(test)]
+mod tests;
+
+#[derive(Debug)]
+pub(crate) struct DropBomb {
```

Comment:
I experimented a bit more with `#[must_use]` and I think that it's not enough, so a drop bomb is a good idea :+1:

Reply:
Yeah, the reason for using drop bombs was two-fold:

1. `#[must_use]` was not sufficient to guarantee that constructed command wrappers are *eventually* actually executed.
2. `#[must_use]` was noisy on intermediate builder methods (especially if a builder method wants to be conditionally called).